src/service/xhr-impl.js

```diff
@@ -189,8 +203,39 @@ export class Xhr {
       init.headers['Content-Type'] = 'application/json;charset=utf-8';
       init.body = JSON.stringify(init.body);
     }
-    return this.fetch(input, init)
-        .then(response => response.json());
+
+    const isCacheable = (init.method === 'GET' && propertyPath);
+    if (isCacheable) {
+      const cacheKey = this.getCacheKey_(input, opt_init);
+      const cachedPromise = this.fragmentCache_.get(cacheKey);
```

Comment:
> The fragment is a path to a property inside a JSON object.

We've defined that for `fetchJSON` only.

But fragments can be added to any request, and they're never sent to the server. So, `fecthDoc("#frag1")` and `fecthDoc("#frag2")` are the same request, but won't be cached properly. They _should_ be cached, without doing anything special to the response (no deep JSON property returning, etc).